lib/open_mail_app.dart

```diff
@@ -80,32 +81,50 @@ class OpenMailApp {
   /// iOS: [MailApp.iosLaunchScheme] will be populated
   static Future<List<MailApp>> getMailApps() async {
     if (Platform.isAndroid) {
-      var appsJson = await _channel.invokeMethod<String>('getMainApps');
-      var apps = <MailApp>[];
-
-      if (appsJson != null) {
-        apps = (jsonDecode(appsJson) as Iterable)
-            .map((x) => MailApp.fromJson(x))
-            .toList();
-      }
-
-      return apps;
+      return await _getAndroidMailApps();
     } else if (Platform.isIOS) {
       return await _getIosMailApps();
     } else {
       throw Exception('Platform not supported');
     }
   }
 
+  static Future<List<MailApp>> _getAndroidMailApps() async {
+    var appsJson = await _channel.invokeMethod<String>('getMainApps');
+    var apps = <MailApp>[];
+
+    if (appsJson != null) {
+      apps = (jsonDecode(appsJson) as Iterable)
+          .map((x) => MailApp.fromJson(x))
+          .toList();
+    }
+
+    apps.removeWhere((MailApp app) {
+      return _filterList.contains(app.name.toLowerCase());
+    });
```

Comment:
I think we could chain a where clause when we first create the `apps` list.

```suggestion
    if (appsJson != null) {
      apps = (jsonDecode(appsJson) as Iterable)
          .map((x) => MailApp.fromJson(x))
          .where((x) => !_filterList.contains(app.name.toLowerCase()))
          .toList();
    }
```